### PR TITLE
CAMS-436: Dropdown menu accessibility

### PR DESCRIPTION
**Pull request by fmaddenflx**
# Purpose

Make sure Dropdown menu follows all accessibility rules

# Major Changes

Rules followed for Dropdown Menu component:

- https://www.w3.org/WAI/ARIA/apg/patterns/menu-button/
- https://www.w3.org/WAI/ARIA/apg/patterns/menubar/

for more information see https://docs.google.com/document/d/1jWB0gHGiKyAIsX-4o5PQQfbxQgs-bMT_wsGOdyEPA2c/edit?tab=t.0

# Testing/Validation

100% test coverage

# Definition of Done:

- [x] Acceptance criteria met - The system meets all acceptance criteria
- [x] Feature toggles created - Features that are deployed but not released have toggles
- [x] Design QA passed - All visual UI looks and works the way it’s supposed to
- [x] Usability validated - Team ensures the working code is easy to use
- [x] Accessibility tests passed - System passes automated and manual accessibility tests and supports applicable devices
- [x] Code refactored for clarity - Developers can understand the work simply by reviewing the code
- [x] Dependency Rule followed - More important code doesn’t directly depend on less important code
- [x] Development debt eliminated - UX and code aligns to the team’s latest understanding of the domain
- [ ] Source code merged - Code has been merged into the main branch
- [x] 90% unit test coverage achieved - Automated test coverage tools indicate coverage of >= 90%
- [x] Code reviewed - Code is reviewed by at least two other team members before being merged
- [x] Code quality checks passed - Code passes all automated quality checks
- [x] Security scans passed - Code passes expected scans for vulnerabilities and compliance issues
- [x] Threat model updated - The threat model incorporates new threats and remediations
- [x] Build process updated - Automated build process include code supporting the user story
- [x] Load/performance tests passed - Performance tests include the functionality of the user story
- [x] Documents are updated - All required documentation is up to date and version controlled
- [x] Code is deployed - Code is deployed to the highest production like environment

### CODE REVIEW

File: user-interface/src/lib/components/cams/DropdownMenu/DropdownMenu.tsx [hunk 1]
```diff
@@ -31,12 +33,11 @@ export function DropdownMenu(props: DropdownMenuProps) {
...
 
-  function handleMenuKeyDown(ev: React.KeyboardEvent) {
-    if (ev.key === 'ArrowDown' && expanded === true) {
-      const firstItem = document.querySelector(`#menu-link-${id}-0`);
-      if (firstItem) (firstItem as HTMLLIElement).focus();
-    } else if (expanded === true && ev.key === 'Tab' && ev.shiftKey === true) {
-      handleToggleExpand();
+  function handleClickOutside(ev: MouseEvent) {
+    const itemList = document.getElementById(`${id}-item-list`);
+    //if (buttonRef.current && !buttonRef.current.contains(ev.target as Node)) {
```
Comment (btposey) [inline user-interface/src/lib/components/cams/DropdownMenu/DropdownMenu.tsx:38]: Dead code.

```suggestion
```

File: user-interface/src/lib/components/cams/DropdownMenu/DropdownMenu.tsx [hunk 2]
```diff
@@ -45,26 +46,50 @@ export function DropdownMenu(props: DropdownMenuProps) {
...
+      const firstItem: HTMLLinkElement | null = document.querySelector(`#menu-link-${id}-0`);
+      const lastItem: HTMLLinkElement | null = document.querySelector(
+        `#menu-link-${id}-${submenuItemCount - 1}`,
+      );
       if (ev.key === 'ArrowDown') {
         if (ev.currentTarget === lastItem) {
-          handleToggleExpand();
-          setFocus(true);
+          if (firstItem) firstItem.focus();
+          ev.preventDefault();
```
Comment (btposey) [inline user-interface/src/lib/components/cams/DropdownMenu/DropdownMenu.tsx:56]: Redundant `preventDefault`
```suggestion
```

File: user-interface/src/lib/components/cams/DropdownMenu/DropdownMenu.tsx [hunk 3]
```diff
@@ -45,26 +46,50 @@ export function DropdownMenu(props: DropdownMenuProps) {
...
           const nextItem = ev.currentTarget.parentElement?.nextElementSibling;
           if (nextItem) (nextItem.children[0] as HTMLLIElement).focus();
         }
+        ev.preventDefault();
       } else if (ev.key === 'ArrowUp') {
         if (ev.currentTarget === firstItem) {
-          handleToggleExpand();
-          setFocus(true);
+          if (lastItem) lastItem.focus();
+          ev.preventDefault();
```
Comment (btposey) [inline user-interface/src/lib/components/cams/DropdownMenu/DropdownMenu.tsx:65]: Another redundant `preventDefault`.
```suggestion
```

**Review by btposey — CHANGES_REQUESTED** [review]

File: user-interface/src/lib/components/Header.scss [hunk 4]
```diff
@@ -240,3 +238,33 @@
...
+        .site-title.wide-screen {
+          display: flex;
+        }
+        .site-title.small-screen {
+          display: none;
+        }
+      }
+    }
+  }
+}
```
Comment (jamesobrooks) [inline user-interface/src/lib/components/Header.scss:270]: Do these widths not have an overlap? It seems like from 73 to 74 em we would have ambiguity as to which style applies.

Comment (fmaddenflx) [reply user-interface/src/lib/components/Header.scss:270]: Great point.

File: user-interface/src/lib/components/cams/DropdownMenu/DropdownMenu.tsx (hunk 1 above)
Comment (jamesobrooks) [inline user-interface/src/lib/components/cams/DropdownMenu/DropdownMenu.tsx:38]: Remove if not needed.
```suggestion
```

File: user-interface/src/lib/components/cams/DropdownMenu/DropdownMenu.tsx [hunk 5]
```diff
@@ -45,26 +46,50 @@ export function DropdownMenu(props: DropdownMenuProps) {
...
         } else {
           const previousItem = ev.currentTarget.parentElement?.previousElementSibling;
           if (previousItem) (previousItem.children[0] as HTMLLIElement).focus();
         }
-      } else if (ev.key === 'Tab' && ev.shiftKey === false) {
+        ev.preventDefault();
```
Comment (jamesobrooks) [inline user-interface/src/lib/components/cams/DropdownMenu/DropdownMenu.tsx:70]: It seems like this makes line 65 unnecessary, or applies to the else block on 66 where maybe we don't want to.

**Review by jamesobrooks — APPROVED** [review]

**Review by btposey — APPROVED** [review]

**Pull request merged**